File: profiling/many_calls_proxy.py
import time
import contextlib

import Pyro4


@contextlib.contextmanager
def timeit(name, formatter=None):
    t0 = time.time()
    yield
    print("{}: {:.3f}ms".format(name, (time.time()-t0)*1000))


n_calls = 100

uri = "PYRO:TestServer@localhost:50001"
p = Pyro4.Proxy(uri)

t0 = time.time()
with timeit("calls"):
    for i in range(100):
        with timeit("call"):
            p.square(2)

File: profiling/many-calls-proxy.js
const { Proxy, locateNS } = require("./../index.js")

var nCalls = 100

var main = async ()=>{

    var uri
    await locateNS(async (ns)=>{
        uri = await ns.lookup(["TestServer"])
        console.log(uri)
    })
    var uri = "PYRO:TestServer@localhost:50001"
    var p = new Proxy(uri)
    await p.init()
    console.time("calls")
    for (var i = 0; i < nCalls; i++){
        console.time("call")
        await p.square([2])
        console.timeEnd("call")
    }
    console.timeEnd("calls")
    await p.end()
}

main()
